assets/src/edit-story/panels/panel/handle.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useRef,
+	useState,
+	useCallback,
+	useLayoutEffect,
+} from '@wordpress/element';
+
+const Handle = styled.button.attrs( { type: 'button' } )`
+	background: transparent;
+	border: 0;
+	padding: 0;
+	position: absolute;
+	top: 0;
+	left: 50%;
+	margin-left: -60px;
+	width: 120px;
+	height: 20px;
+	display: flex;
+	flex-direction: column;
+	justify-content: flex-start;
+	align-items: center;
+	cursor: row-resize;
+`;
+
+const Bar = styled.div`
+	margin-top: 4px;
+	background-color: ${ ( { theme } ) => theme.colors.bg.v0 };
+	width: 32px;
+	height: 4px;
+	border-radius: 2px;
+`;
+
+function DragHandle( { handleHeightChange } ) {
+	const handle = useRef();
+	const lastPosition = useRef();
+	const [ isDragging, setIsDragging ] = useState( false );
+
+	// On mouse move, check difference since last record vertical mouse position
+	// and invoke callback with this difference.
+	// Then record new vertical mouse position for next iteration.
+	const handleMouseMove = useCallback( ( evt ) => {
+		const delta = lastPosition.current - evt.pageY;
+		handleHeightChange( delta );
+		lastPosition.current = evt.pageY;
+	}, [ handleHeightChange ] );
+
+	// On mouse up, set dragging as false
+	// - will cause useLayoutEffect to unregister listeners.
+	const handleMouseUp = useCallback( () => setIsDragging( false ), [] );
+
+	// On mouse down, set dragging as true
+	// - will cause useLayoutEffect to register listeners.
+	// Also record the initial vertical mouse position on the page.
+	const handleMouseDown = useCallback( ( evt ) => {
+		lastPosition.current = evt.pageY;
+		setIsDragging( true );
+	}, [] );
+
+	// On initial render *and* every time `isDragging` changes value,
+	// register all relevant listeners. Note that all listeners registered
+	// will be correctly unregistered due to the cleanup function.
+	useLayoutEffect( () => {
+		const element = handle.current;
+		const doc = element.ownerDocument;
+		element.addEventListener( 'mousedown', handleMouseDown );
+
+		if ( isDragging && doc ) {
+			doc.addEventListener( 'mousemove', handleMouseMove );
+			doc.addEventListener( 'mouseup', handleMouseUp );
+		}
+
+		return () => {
+			if ( element ) {
+				element.removeEventListener( 'mousedown', handleMouseDown );
+				if ( isDragging && doc ) {
+					doc.removeEventListener( 'mousemove', handleMouseMove );
+					doc.removeEventListener( 'mouseup', handleMouseUp );
+				}
+			}
+		};
+	}, [ isDragging, handleMouseUp, handleMouseMove, handleMouseDown ] );
+
+	return (
+		<Handle ref={ handle }>
```

Comment:
What accessibility could we add here? CC @pbakaus as he has done a lot of accessibility with drag and drop.

Reply:
My thinking:

* Keyboard shortcuts to go toggle resize mode and do resizing. Example:
  <kbd>Space</kbd> to toggle resize mode, <kbd>Up</kbd> and <kbd>Down</kbd> to resize
* Screenreader messages (`aria-live`) to announce state changes

Reply:
So, I've done it as follows:
https://github.com/ampproject/amp-wp/blob/cebef7544bb234659f53e39379355f2ea1682ced/assets/src/edit-story/panels/panel/handle.js#L46-L56

Plus it has a keyboard handler reacting to up and down arrow presses, that then change the height up and down.

This is partly informed by [this question](https://stackoverflow.com/questions/42981485/proper-accessibility-or-aria-role-for-a-resize-handle), but I'm using `role="slider"` actively going against one of the recommendations there, as it allows me to announce the range and current value with other attributes. Thus I don't think it needs `aria-live`, which could be a bit weird here.

I was considering whether to use `aria-valuetext` to announce the height in relative terms - e.g. `"Panel is 55% of maximum height"` or something similar. That might be a bit too much?

I've also included the information about what the button does as actual textual content of the button (but hidden it for browser display with text-indent). I tihnk this is recommended over using `aria-label`.